### PR TITLE
Fixes a lot of spy-thief uplink jank

### LINKED ISSUE #12287: Medis Satcyheoll cannot be collected
### Labels

[BUG][MINOR]

### Description

can't collect it as a spief

### Steps to reproduce

try to collect it

### Expected Behavior

to collect it

### Additional Information & Screenshots

Reported by: LadyGeartheart
On server: Goonstation 3 Roleplay: Morty
Round log date: 2022-12-04-09-04
Reported on: 2022-12-04 10:10:33
Map: DONUT3

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[BUG] [QOL] [GAMEMODES]
## About the PR 
Makes spy thief uplinks use `COMSIG_ITEM_ATTACKBY_PRE` to detect hitting an object, allowing them to cancel the `attackby` call if they are able to deliver it. This means hitting a bag or table in the correct area will now start the delivery action bar instead of putting your PDA on/in it.
Fixes the spy thief GPS system not working unless you click dragged, so now the funny blue line appears much more consistently. (still has issues with armoury doors and some other stuff though)
Removes the click-drag delivery option as it should no longer be necessary.

## Why's this needed? 
Spy thief delivery is currently buggy and inconsistent, this fixes that.
Fixes #12287

## Changelog 

```changelog
(u)LeahTheTech
(*)Spy-thief delivery teleportation should now work regardless of what you are clicking on (bags, tables etc.)
(+)Spy-thief area GPS now works again
```

### CODE REVIEW

File: code/obj/item/uplinks.dm [hunk 1]
```diff
@@ -820,99 +831,103 @@ Note: Add new traitor items to syndicate_buylist.dm, not here.
...
 
-		user.show_text("You cannot claim [delivery] for bounty!", "red")
+		var/datum/bounty_claim/claim = src.bounty_is_claimable(delivery)
+		if (!claim)
+			user.show_text("You cannot claim [delivery] for bounty!", "red")
+			src.ui_update()
+			return FALSE
+		var/datum/bounty_item/bounty = claim.bounty
+		delivery = claim.delivery
+		user.removeGpsPath(doText = 0)
```
Comment (frawhst) [inline code/obj/item/uplinks.dm:896]:
```suggestion
		user.removeGpsPath(doText = FALSE)
```

File: code/obj/item/uplinks.dm [hunk 2]
```diff
@@ -820,99 +831,103 @@ Note: Add new traitor items to syndicate_buylist.dm, not here.
...
+		bounty.claimed = TRUE
+		if (istype(delivery.loc, /mob))
+			var/mob/M = delivery.loc
+			if (istype(delivery,/obj/item/parts) && ishuman(M))
+				var/mob/living/carbon/human/H = M
+				var/obj/item/parts/HP = delivery
+			//	var/limb_name = HP.holder.real_name + "'s " + HP.name
```
Comment (frawhst) [inline code/obj/item/uplinks.dm:903]:
```suggestion
```

**Review by frawhst — APPROVED** [review]
couple nitpicks otherwise lgtm

File: code/obj/item/uplinks.dm [hunk 3]
```diff
@@ -779,6 +779,16 @@ Note: Add new traitor items to syndicate_buylist.dm, not here.
 	spy
 		purchase_flags = UPLINK_SPY
 
+///Datum used to combine the bounty being claimed with the item being delivered
+/datum/bounty_claim
+	var/datum/bounty_item/bounty = null
+	var/atom/delivery = null
+
+	New(datum/bounty_item/bounty, atom/delivery)
```
Comment (ZeWaka) [inline code/obj/item/uplinks.dm:787]: disposing

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/item/uplinks.dm (hunk 3 above)
Comment (TobleroneSwordfish) [reply code/obj/item/uplinks.dm:787]: Hope that's all I need to do? I don't want to dispose the properties as this is just a datum for combining pre-existing arguments into one return.

**Pull request merged**